service/test/BatchServerTest.cpp

```diff
@@ -285,34 +264,28 @@ TEST_F(BatchServerTest, run_batch_read_empty)
         .RetiresOnSaturation();
 
     EXPECT_CALL(*m_batch_status,
-                send_message(BatchStatus::M_MESSAGE_CONTINUE))
+                send_message(BatchStatus::M_MESSAGE_QUIT))
         .Times(1)
         .RetiresOnSaturation();
 
-    m_batch_server->run_batch();
-
-    EXPECT_EQ(result.size(), m_signal_shmem->size());  // both should be == 0
+    m_batch_server_controls->run_batch();
```

Comment:
Hmm ... do we want this one to be `run_batch()` or `event_loop()`? It seems that we aren't handling any theoretical thrown exceptions in this particular unit test. I assume that these exceptions can occur upon receipt of any signal other than `SIGTERM`? Maybe it can be used to test the failure condition?

Reply:
Would be great to get better testing of cases where mock objects in the unit tests are made to throw exceptions when they are called.